authentication/pipeline/user.py

```diff
@@ -30,6 +33,55 @@
 # pylint: disable=keyword-arg-before-vararg
 
 
+def forbid_hijack(strategy, backend, **kwargs):  # pylint: disable=unused-argument  # noqa: ARG001
+    """
+    Forbid an admin user from trying to login/register while hijacking another user
+
+    Args:
+        strategy (social_django.strategy.DjangoStrategy): the strategy used to authenticate
+        backend (social_core.backends.base.BaseAuth): the backend being used to authenticate
+    """
+    # As first step in pipeline, stop a hijacking admin from going any further
+    if bool(strategy.session_get("hijack_history")):
+        raise AuthException("You are hijacking another user, don't try to login again")  # noqa: EM101
+    return {}
+
+
+# Pipeline steps for OIDC logins
+
+
+def create_ol_oidc_user(strategy, details, backend, user=None, *args, **kwargs):
+    """
+    Create the user if we're using the ol-oidc backend.
+
+    This also does a blocked user check and makes sure there's an email address.
+    If the created user is new, we make sure they're set active. (If the user is
+    inactive, they'll get knocked out of the pipeline elsewhere.)
+    """
+
+    if backend.name != OlOpenIdConnectAuth.name:
+        return {}
+
+    if "email" not in details:
+        raise RequireEmailException(backend, None)
+
+    if "email" in details and is_user_email_blocked(details["email"]):
+        raise EmailBlockedException(backend, None)
+
+    retval = create_user(strategy, details, backend, user, *args, **kwargs)
+
+    if retval.get("is_new"):
+        # the backend should be setting the is_active flag to True (and it seems
+        # to be doing that!) but it isn't so we need to set it to active here.
```

Comment:
Once we switchover completely to keycloak, we can set the default on the `User` model to `True` and remove this code.